src/FormBuilder/BlockModels/DonationAmountBlockModel.php

```diff
@@ -60,23 +60,22 @@ public function getLabel(): string
     }
 
     /**
+     * @unreleased Changed the return type to an array of OptionsProps
      * @since 3.0.0
      *
-     * @return float[]
+     * @return array ['label' => string, 'value' => string, 'checked' => bool][]
      */
     public function getLevels(): array
     {
-        return array_map(static function($level) {
-            return (float)filter_var($level, FILTER_SANITIZE_NUMBER_FLOAT,FILTER_FLAG_ALLOW_FRACTION);
-        }, $this->block->getAttribute('levels'));
```

Comment:
@JoshuaHungDinh @pauloiankoski was removing this filtering intentional? Remember the `getAttribute()` method just returns whatever is in the database underneath that attribute key.  That means we are relying on the data to provide the type which sounds fine in theory but is not something I recommended with blocks.  This can be problematic which is one of the reasons I created the [BlockType](https://github.com/impress-org/givewp/blob/becc18a76b646945c5a62b698609eaca733b72bc/src/Framework/Blocks/BlockType.php) so we can validate and cast our attributes to the data source we expect.

I would recommend casting the `levels` attribute in the shape you are expecting and to continue using the filter_var approach when necessary

Reply:
Would you mean like this?
![CleanShot 2024-04-29 at 19 37 00](https://github.com/impress-org/givewp/assets/3921017/3f5e60bc-18c3-408b-982b-56c1627d69bf)

About the BlockType, did I get it correctly that it may be a replacement to the custom BlockModels we have now? I got little bit confused since they have a lot of similarities.

Reply:
@pauloiankoski yes on your example, that's what I was imagining 👍 

Re: BlockType, yes the custom BlockModel we have for the donation amount will eventually be replaced by a BlockType.  That was my first attempt before creating a system to be re-used.  I plan on working on that migration during a fun friday 😄

Reply:
Resolved on https://github.com/impress-org/givewp/pull/7364/commits/48ba42be90668b55354a11d4bba37bfc478deb9c

Reply:
addressing a test: 99322ff8163936c17299142b60fffb7acba5dfbe